Bottom: c03eb3b711ec513a76e2c761c7a81a25c455cdbc
Top:    06b63c25ea7eaba03b26b6e5ce9039aa9a31aff4
Author: Ricard Salveti <ricardo.salveti@linaro.org>
Date:   2016-09-23 18:43:13 +0530

From 8e9604391898e292ea4d3f74fde1d19f35b17530 Mon Sep 17 00:00:00 2001
Subject: [PATCH 12/21] boards: 96b_carbon_nrf51: add support for 96Boards
 Carbon nRF51 chip

This adds support for the nRF51 chip on the board. If you'd like to run
zephyr on the STM32F4 chip on Carbon, you need to use the board support
specifically for STM32.

Change-Id: I0cc6a6961c43b9e0c8ffae3a42fbcc089b090803
Signed-off-by: Ricardo Salveti <ricardo.salveti@linaro.org>
Signed-off-by: Amit Kucheria <amit.kucheria@linaro.org>
Signed-off-by: Marti Bolivar <marti.bolivar@linaro.org>


---

diff --git a/boards/arm/96b_carbon_nrf51/96b_carbon_nrf51_defconfig b/boards/arm/96b_carbon_nrf51/96b_carbon_nrf51_defconfig
new file mode 100644
index 000000000..e71e031d6
--- /dev/null
+++ b/boards/arm/96b_carbon_nrf51/96b_carbon_nrf51_defconfig
@@ -0,0 +1,32 @@
+CONFIG_ARM=y
+CONFIG_SOC_FAMILY_NRF5=y
+CONFIG_SOC_SERIES_NRF51X=y
+CONFIG_SOC_NRF51822_QFAC=y
+CONFIG_BOARD_96B_CARBON_NRF51=y
+
+# TODO: Disable console/uart
+# enable uart driver
+CONFIG_SERIAL=y
+CONFIG_UART_NRF5=y
+CONFIG_UART_NRF5_GPIO_TX_PIN=9
+CONFIG_UART_NRF5_GPIO_RX_PIN=11
+
+# enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# bluetooth
+CONFIG_BLUETOOTH=y
+CONFIG_BLUETOOTH_CONTROLLER=y
+
+# spi
+CONFIG_SPI=y
+CONFIG_SPI_1=y
+CONFIG_SPI_1_NAME="SPI_1"
+CONFIG_SPI_1_IRQ_PRI=1
+CONFIG_SPI_NRF5=y
+CONFIG_SPIS1_NRF5=y
+CONFIG_SPIS1_NRF5_GPIO_SCK_PIN=7
+CONFIG_SPIS1_NRF5_GPIO_MOSI_PIN=0
+CONFIG_SPIS1_NRF5_GPIO_MISO_PIN=30
+CONFIG_SPIS1_NRF5_GPIO_CSN_PIN=25
diff --git a/boards/arm/96b_carbon_nrf51/Kconfig.board b/boards/arm/96b_carbon_nrf51/Kconfig.board
new file mode 100644
index 000000000..78602b9c1
--- /dev/null
+++ b/boards/arm/96b_carbon_nrf51/Kconfig.board
@@ -0,0 +1,19 @@
+# Kconfig - 96Boards Carbon (nRF51) board configuration
+#
+# Copyright (c) 2016 Linaro Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+config BOARD_96B_CARBON_NRF51
+	bool "96Boards Carbon (nRF51)"
+	depends on SOC_NRF51822_QFAC
diff --git a/boards/arm/96b_carbon_nrf51/Kconfig.defconfig b/boards/arm/96b_carbon_nrf51/Kconfig.defconfig
new file mode 100644
index 000000000..27efd3894
--- /dev/null
+++ b/boards/arm/96b_carbon_nrf51/Kconfig.defconfig
@@ -0,0 +1,35 @@
+# Kconfig - 96Boards Carbon (nRF51) board configuration
+#
+# Copyright (c) 2016 Linaro Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if BOARD_96B_CARBON_NRF51
+
+config BOARD
+	default 96b_carbon_nrf51
+
+if UART_NRF5
+
+config GPIO
+	default y
+
+config GPIO_NRF5
+	default y
+
+config GPIO_NRF5_P0
+	default y
+
+endif # UART_NRF5
+
+endif # BOARD_96B_CARBON_NRF51
diff --git a/boards/arm/96b_carbon_nrf51/Makefile b/boards/arm/96b_carbon_nrf51/Makefile
new file mode 100644
index 000000000..c925263c4
--- /dev/null
+++ b/boards/arm/96b_carbon_nrf51/Makefile
@@ -0,0 +1,2 @@
+# No C files (yet)
+obj- += dummy.o
diff --git a/boards/arm/96b_carbon_nrf51/board.h b/boards/arm/96b_carbon_nrf51/board.h
new file mode 100644
index 000000000..a793beb55
--- /dev/null
+++ b/boards/arm/96b_carbon_nrf51/board.h
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2016 Linaro Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __INC_BOARD_H
+#define __INC_BOARD_H
+
+#include <soc.h>
+
+#endif /* __INC_BOARD_H */
